admin-dev/themes/new-theme/js/pages/customer/CustomerForm.ts

```diff
@@ -31,9 +31,54 @@ import ChangePasswordHandler from '../../components/change-password-handler';
  */
 export default class CustomerForm {
   constructor() {
+    // Watch password field change and strength indicator
     const passwordHandler = new ChangePasswordHandler(
       customerFormMap.passwordStrengthFeedbackContainer,
     );
     passwordHandler.watchPasswordStrength($(customerFormMap.passwordInput));
+
+    // Watch customer group checkbox change and if it was unchecked,
+    // update default group below if it's no longer in the list.
+    $('input[type="checkbox"][name="customer[group_ids][]"]').on('change', (event) => {
+      this.checkOrUpdateDefaultGroup($(event.currentTarget).is(':checked'));
+    });
+  }
+
+  private checkOrUpdateDefaultGroup(wasChecked: boolean): void {
+    // Get currently selected group ID
+    const currentDefaultGroup = Number($('#customer_default_group_id option:selected').val() as string);
+
+    // Get all checked groups in group access
+    const checkedGroups: number[] = [];
+    let firstGroupInList: number = 0;
+    $('input[type="checkbox"][name="customer[group_ids][]"]').each((index, value) => {
+      // We will keep track of all checked groups
+      if ($(value).is(':checked')) {
+        checkedGroups.push(Number($(value).val() as string));
+      }
+      // And store ID of the first group regardless of it's status
+      if (index === 0) {
+        firstGroupInList = Number($(value).val() as string);
```

Comment:
following above, now you can reuse the var here 
```suggestion
        firstGroupInList = value;
```

Reply:
Yes, but I don't need it all the time, so why construct it?

Reply:
because assigning a var probably costs same or less (performance wise) as extracting the .val() from input and you can reuse it